internal/tofu/node_resource_import.go

```diff
@@ -30,6 +31,13 @@ type graphNodeImportState struct {
 	states []providers.ImportedResource
 }
 
+func (n *graphNodeImportState) SetPotentialProviders(potentialProviders []distinguishableProvider) {
+	// Technically we shouldn't get here as we don't support import with for_each on providers in resources / modules
+	// So we should panic here (ensure we have a code that validates it somewhere else) / implement a proper user error.
+	//TODO Ronny - don't forget to handle this scenario ^
+	panic("implement me")
```

Comment:
We need to ensure the user is notified with a proper error that this code path is currently not possible. In stage one, we won't support using the `import` block to import into resources with a `for_each` on providers in the resource and/or modules containing this resource.

Reply:
Just a note: we have a validation for this case as a part of configuration parsing.